### PR TITLE
Add remote uri support

### LINKED ISSUE #1196: [Bug] Visual Studio Code extension not working
**Describe the bug**
Visual Studio code extension not finding any results.

**To Reproduce**
enable VSC extension, click refresh.

Debug log reads:

```
[9/11/2024, 1:58:38 PM][ERROR] Failed to get search result items for extension with id 'VSCode.Reason: TypeError [ERR_INVALID_URL_SCHEME]: The URL must be of scheme file'
```

The first entry in my history.recentlyOpenedPathsList in the ItemTable has (let me know if you need the full dump):

```JSON
{
    "entries": [
        {
            "folderUri": "file:///c%3A/Users/jeffm/scoop/buckets/milnak"
        }
```

Was this extension tested on Windows?

**Expected behavior**
extension works.

**Environment**
 - OS: Windows 
 - Version 9.5.0 (euli version)

### PR DESCRIPTION
**Pull request by IrishBruse**
Add remote uri support which should fix #1196
Also added some logging to help debug any issues in the future of generating the items

image

### CODE REVIEW

File: src/main/Extensions/VSCode/VSCodeExtension.ts [hunk 1]
```diff
@@ -44,20 +46,25 @@ export class VSCodeExtension implements Extension {
...
-            recentPaths.map((recent) => {
-                return this.getSearchItem(recent);
-            }),
-        );
+        for (const recent of this.getRecents()) {
+            try {
+                searchResults.push(await this.getSearchItem(recent));
+            } catch (error) {
+                const uri = this.getUri(recent);
+                this.logger.error("VSCODE: " + uri + " " + error);
```
Comment (oliverschwendener) [inline src/main/Extensions/VSCode/VSCodeExtension.ts:62]: Please use the extension id here:

```suggestion
                this.logger.error(this.id + ": " + uri + " " + error);
```

File: src/main/Extensions/VSCode/VSCodeExtension.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { searchFilter } from "@common/Core/Search/SearchFilter";
 import Database from "better-sqlite3";
 import * as Path from "path";
 import * as Url from "url";
+import type { Logger } from "../../Core/Logger";
```
Comment (oliverschwendener) [inline src/main/Extensions/VSCode/VSCodeExtension.ts:12]:
```suggestion
import type { Logger } from "@Core/Logger";
```

Comment (IrishBruse) [reply src/main/Extensions/VSCode/VSCodeExtension.ts:12]: Is there any way to get vscode to automatically pick these instead?

Comment (oliverschwendener) [reply src/main/Extensions/VSCode/VSCodeExtension.ts:12]: I don't know, I always get the suggestions.

Comment (IrishBruse) [reply src/main/Extensions/VSCode/VSCodeExtension.ts:12]: Oh interesting i always get relative paths which is annoying must be some setting or extension related thingimage

Comment (IrishBruse) [reply src/main/Extensions/VSCode/VSCodeExtension.ts:12]: Ah found it had to set this "typescript.preferences.importModuleSpecifier": "non-relative",

**Review by oliverschwendener — APPROVED** [review]

**Pull request merged**